### PR TITLE
fix re2 compilation problem 

### PR DESCRIPTION
**Pull request by laurentsimon**
see this issue https://github.com/google/fuzzbench/issues/355

### CODE REVIEW

File: fuzzers/aflplusplus_noalloc/fuzzer.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 Google LLC
```
Comment (jonathanmetzman) [inline fuzzers/aflplusplus_noalloc/fuzzer.py:1]: THink this probably needs to be removed.

**Comment by laurentsimon** [comment]
I'm confused where this file comes from. master branch does not have it.
Yet I pulled from master and here it is. Any ideas?

**Comment by inferno-chromium** [comment]
> I'm confused where this file comes from. master branch does not have it. Yet I pulled from master and here it is. Any ideas?

unsure why. i just deleted the file from CL using github ui, so we should be good, and you can try creating a new branch from master to see if file still persists.

**Comment by inferno-chromium** [comment]
Still fails, can you try building this on a x509 benchmark

```
Running split-switches-pass by laf.intel@gmail.com
Rewriting 1 switch statements 
Running compare-transform-pass by laf.intel@gmail.com
Split-compare-pass by laf.intel@gmail.com
Running split-compare-pass 64
Running split-compare-pass 32
Running split-compare-pass 16
make[1]: Leaving directory '/src/openssl'
Makefile:2214: recipe for target 'build_sw' failed
make: *** [build_sw] Error 2
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/src/fuzzer.py", line 53, in build
    utils.build_benchmark()
  File "/src/fuzzers/utils.py", line 52, in build_benchmark
    subprocess.check_call(['/bin/bash', '-ex', build_script], env=env)
  File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/bin/bash', '-ex', '/src/build.sh']' returned non-zero exit status 2
The command '/bin/sh -c PYTHONPATH=$SRC python3 -u -c "from fuzzers import utils; utils.initialize_flags(); import fuzzer; fuzzer.build()"' returned a non-zero code: 1
make: *** [.lafintel-openssl_x509-oss-fuzz-builder] Error 1
docker/build.mk:303: recipe for target '.lafintel-openssl_x509-oss-fuzz-builder' failed
Building benchmarks: openssl_x509, bloaty_fuzz_target, sqlite3_ossfuzz, zlib_zlib_uncompress_fuzzer, curl_curl_fuzzer_http, systemd_fuzz-link-parser, jsoncpp_jsoncpp_fuzzer, mbedtls_fuzz_dtlsclient, libpcap_fuzz_both for fuzzer: lafintel
Running command: make RUNNING_ON_CI=yes -j test-run-lafintel-openssl_x509
```

if your git branch is bad, can try a new PR.

**Comment by inferno-chromium** [comment]
re2 itself fails

```
/usr/local/include/c++/v1/new:233:33: error: use of undeclared identifier 'max_align_t'
  return __align > alignment_of<max_align_t>::value;
                                ^
In file included from util/arena.cc:5:
In file included from ./util/util.h:20:
In file included from /usr/local/include/c++/v1/vector:276:
In file included from /usr/local/include/c++/v1/__bit_reference:15:
In file included from /usr/local/include/c++/v1/algorithm:643:
In file included from /usr/local/include/c++/v1/memory:671:
/usr/local/include/c++/v1/new:233:33: error: use of undeclared identifier 'max_align_t'
  return __align > alignment_of<max_align_t>::value;
                                ^
The command '/bin/sh -c python3 -u -c "from fuzzers import utils; utils.initialize_flags(); import fuzzer; fuzzer.build()"' returned a non-zero code: 1
1 error generated.
```

**Comment by laurentsimon** [comment]
I tested this [commit](https://github.com/google/fuzzbench/pull/356/commits/b9cdb2927465ebdacd3f7e30077e13beb9548aa3) on my machine and it worked.

Is this error from an earlier commit?

**Comment by inferno-chromium** [comment]
> I tested this [commit](https://github.com/google/fuzzbench/pull/356/commits/b9cdb2927465ebdacd3f7e30077e13beb9548aa3) on my machine and it worked.
> 
> Is this error from an earlier commit?

Nope, since i see this in Github Actions which has your `-std=c11`

+ CXXFLAGS='-stdlib=libc++ -pthread -Wl,--no-as-needed -Wl,-ldl -Wl,-lm -Wno-unused-command-line-argument -O3 -fPIC -std=c11 -I/usr/local/include/c++/v1/ -std=c++11 -std=gnu++98'

**Comment by laurentsimon** [comment]
I'm unable to reproduce this problem locally. I use lafintel3 branch on cloudtop and it compiles without errors.

**Comment by inferno-chromium** [comment]
This happens for me locally as well. Can you try on a new branch, git pull, do you have any other changes on your branch.

```
/usr/local/include/c++/v1/new:233:33: error: use of undeclared identifier 'max_align_t'
  return __align > alignment_of<max_align_t>::value;
                                ^
In file included from re2/perl_groups.cc:4:
In file included from ./re2/unicode_groups.h:21:
In file included from ./util/util.h:20:
In file included from /usr/local/include/c++/v1/vector:276:
In file included from /usr/local/include/c++/v1/__bit_reference:15:
In file included from /usr/local/include/c++/v1/algorithm:643:
In file included from /usr/local/include/c++/v1/memory:671:
/usr/local/include/c++/v1/new:233:33: error: use of undeclared identifier 'max_align_t'
  return __align > alignment_of<max_align_t>::value;
                                ^
Running split-switches-pass by laf.intel@gmail.com
Running compare-transform-pass by laf.intel@gmail.com
Split-compare-pass by laf.intel@gmail.com
Running split-compare-pass 64
Running split-compare-pass 32
Running split-compare-pass 16
Running split-switches-pass by laf.intel@gmail.com
Running compare-transform-pass by laf.intel@gmail.com
Split-compare-pass by laf.intel@gmail.com
Running split-compare-pass 64
Running split-compare-pass 32
Running split-compare-pass 16
1 error generated.
1 error generated.
Makefile:168: recipe for target 'obj/so/util/stringpiece.o' failed
make: *** [obj/so/util/stringpiece.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [obj/so/util/valgrind.o] Error 1
Makefile:168: recipe for target 'obj/so/util/valgrind.o' failed
1 error generated.
Makefile:160: recipe for target 'obj/util/strutil.o' failed
make: *** [obj/util/strutil.o] Error 1
1 error generated.
1 error generated.
Makefile:160: recipe for target 'obj/re2/prog.o' failed
make: *** [obj/re2/prog.o] Error 1
make: *** [obj/util/valgrind.o] Error 1
Makefile:160: recipe for target 'obj/util/valgrind.o' failed
1 error generated.
1 error generated.
make: *** [obj/re2/parse.o] Error 1
Makefile:160: recipe for target 'obj/re2/parse.o' failed
make: *** [obj/so/re2/bitstate.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/bitstate.o' failed
1 error generated.
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/parse.o' failed
make: *** [obj/so/re2/parse.o] Error 1
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/prog.o' failed
make: *** [obj/so/re2/prog.o] Error 1
Makefile:160: recipe for target 'obj/re2/set.o' failed
make: *** [obj/re2/set.o] Error 1
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/prefilter.o' failed
make: *** [obj/so/re2/prefilter.o] Error 1
1 error generated.
Makefile:168: recipe for target 'obj/so/util/arena.o' failed
make: *** [obj/so/util/arena.o] Error 1
1 error generated.
Makefile:160: recipe for target 'obj/re2/prefilter.o' failed
make: *** [obj/re2/prefilter.o] Error 1
1 error generated.
Makefile:168: recipe for target 'obj/so/util/stringprintf.o' failed
make: *** [obj/so/util/stringprintf.o] Error 1
1 error generated.
1 error generated.
make: *** [obj/util/arena.o] Error 1
Makefile:160: recipe for target 'obj/util/arena.o' failed
Makefile:160: recipe for target 'obj/util/stringprintf.o' failed
make: *** [obj/util/stringprintf.o] Error 1
1 error generated.
Makefile:160: recipe for target 'obj/re2/perl_groups.o' failed
make: *** [obj/re2/perl_groups.o] Error 1
1 error generated.
make: *** [obj/so/util/hash.o] Error 1
Makefile:168: recipe for target 'obj/so/util/hash.o' failed
1 error generated.
Makefile:160: recipe for target 'obj/util/hash.o' failed
1 error generated.
make: *** [obj/util/hash.o] Error 1
1 error generated.
Makefile:160: recipe for target 'obj/re2/unicode_casefold.o' failed
make: *** [obj/re2/unicode_casefold.o] Error 1
1 error generated.
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/simplify.o' failed
make: *** [obj/so/re2/simplify.o] Error 1
make: *** [obj/so/re2/unicode_groups.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/unicode_groups.o' failed
make: *** [obj/so/util/strutil.o] Error 1
Makefile:168: recipe for target 'obj/so/util/strutil.o' failed
1 error generated.
1 error generated.
1 error generated.
1 error generated.
1 error generated.
make: *** [obj/re2/mimics_pcre.o] Error 1
Makefile:160: recipe for target 'obj/re2/mimics_pcre.o' failed
Makefile:160: recipe for target 'obj/re2/dfa.o' failed
make: *** [obj/re2/dfa.o] Error 1
make: *** [obj/re2/unicode_groups.o] Error 1
make: *** [obj/so/re2/perl_groups.o] Error 1
Makefile:160: recipe for target 'obj/re2/unicode_groups.o' failed
Makefile:168: recipe for target 'obj/so/re2/perl_groups.o' failed
Makefile:168: recipe for target 'obj/so/re2/unicode_casefold.o' failed
make: *** [obj/so/re2/unicode_casefold.o] Error 1
1 error generated.
1 error generated.
make: *** [obj/re2/simplify.o] Error 1
Makefile:160: recipe for target 'obj/re2/simplify.o' failed
1 error generated.
make: *** [obj/re2/filtered_re2.o] Error 1
Makefile:160: recipe for target 'obj/re2/filtered_re2.o' failed
1 error generated.
Makefile:160: recipe for target 'obj/util/stringpiece.o' failed
make: *** [obj/util/stringpiece.o] Error 1
Makefile:160: recipe for target 'obj/re2/bitstate.o' failed
make: *** [obj/re2/bitstate.o] Error 1
1 error generated.
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/onepass.o' failed
make: *** [obj/so/re2/onepass.o] Error 1
Makefile:160: recipe for target 'obj/re2/prefilter_tree.o' failed
make: *** [obj/re2/prefilter_tree.o] Error 1
1 error generated.
1 error generated.
make: *** [obj/so/re2/set.o] Error 1
make: *** [obj/re2/compile.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/set.o' failed
Makefile:160: recipe for target 'obj/re2/compile.o' failed
1 error generated.
1 error generated.
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/filtered_re2.o' failed
make: *** [obj/so/re2/filtered_re2.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/mimics_pcre.o' failed
make: *** [obj/so/re2/mimics_pcre.o] Error 1
1 error generated.
Makefile:160: recipe for target 'obj/re2/tostring.o' failed
make: *** [obj/re2/tostring.o] Error 1
1 error generated.
Makefile:160: recipe for target 'obj/re2/nfa.o' failed
make: *** [obj/re2/nfa.o] Error 1
1 error generated.
1 error generated.
1 error generated.
Makefile:160: recipe for target 'obj/re2/onepass.o' failed
make: *** [obj/re2/onepass.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/regexp.o' failed
make: *** [obj/so/re2/regexp.o] Error 1
1 error generated.
Makefile:160: recipe for target 'obj/re2/re2.o' failed
make: *** [obj/re2/re2.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/nfa.o' failed
make: *** [obj/so/re2/nfa.o] Error 1
make: *** [obj/so/re2/compile.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/compile.o' failed
1 error generated.
Makefile:168: recipe for target 'obj/so/re2/re2.o' failed
make: *** [obj/so/re2/re2.o] Error 1
1 error generated.
1 error generated.
Makefile:160: recipe for target 'obj/re2/regexp.o' failed
make: *** [obj/re2/regexp.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/tostring.o' failed
make: *** [obj/so/re2/tostring.o] Error 1
1 error generated.
make: *** [obj/so/re2/prefilter_tree.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/prefilter_tree.o' failed
1 error generated.
make: *** [obj/so/re2/dfa.o] Error 1
Makefile:168: recipe for target 'obj/so/re2/dfa.o' failed
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/src/fuzzer.py", line 55, in build
    utils.build_benchmark()
  File "/src/fuzzers/utils.py", line 52, in build_benchmark
    subprocess.check_call(['/bin/bash', '-ex', build_script], env=env)
  File "/usr/local/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/bin/bash', '-ex', 'benchmark/build.sh']' returned non-zero exit status 1.
The command '/bin/sh -c python3 -u -c "from fuzzers import utils; utils.initialize_flags(); import fuzzer; fuzzer.build()"' returned a non-zero code: 1
make: *** [docker/build.mk:181: .lafintel-re2-2014-12-09-builder] Error 1
(.venv) aarya@aarya:/build/fuzzbench$ git diff
diff --git a/fuzzers/lafintel/fuzzer.py b/fuzzers/lafintel/fuzzer.py
index b8d5b0d..00953ea 100644
--- a/fuzzers/lafintel/fuzzer.py
+++ b/fuzzers/lafintel/fuzzer.py
@@ -30,6 +30,8 @@ def prepare_build_environment():
     cflags = ['-fPIC']
     if 'php' in os.environ['BENCHMARK']:
         cflags += ['-D__builtin_cpu_supports\\(x\\)=0']
+    if 're2' in os.environ['BENCHMARK']:
+        cflags += ['-std=c11']
     cppflags = cflags + ['-I/usr/local/include/c++/v1/', '-std=c++11']
     utils.append_flags('CFLAGS', cflags)
     utils.append_flags('CXXFLAGS', cppflags)
```

**Comment by inferno-chromium** [comment]
maybe clear your docker image cache, you might be on old clang image, as clang was recently updated in base-builder (from oss-fuzz).

**Comment by laurentsimon** [comment]
Thanks for the suggestion. I did manage to reproduce the problem after clearing the cache.

File: benchmarks/re2-2014-12-09/build.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ apt-get update && \
   automake
 
 
-CXXFLAGS="${CXXFLAGS} -std=gnu++98"
+CXXFLAGS="${CXXFLAGS}"
```
Comment (inferno-chromium) [inline benchmarks/re2-2014-12-09/build.sh:25]: This is a no-op, should be removed.

File: fuzzers/lafintel/fuzzer.py [hunk 3]
```diff
@@ -44,7 +43,7 @@ def prepare_build_environment():
     os.environ['CC'] = '/afl/afl-clang-fast'
     os.environ['CXX'] = '/afl/afl-clang-fast++'
     os.environ['FUZZER_LIB'] = '/libAFL.a'
-
+
```
Comment (inferno-chromium) [inline fuzzers/lafintel/fuzzer.py:46]: Lets remove these extra spaces.

**Pull request merged**